src/lib/mina/account-update.ts

```diff
@@ -1171,7 +1197,10 @@ class AccountUpdate implements Types.AccountUpdate {
   static empty() {
     return AccountUpdate.dummy();
   }
-  static check = Types.AccountUpdate.check;
+  static check = (au: AccountUpdate) => {
+    Types.AccountUpdate.check(au);
+    Types.MayUseToken.check(au.body.mayUseToken);
+  };
```

Comment:
this override is not necessary, `Types.AccountUpdate.check()` already works by recursively calling check on all the types it is composed of